recipes-sota/aktualizr/garage-sign-version.inc

```diff
@@ -0,0 +1,23 @@
+
+python () {
+    if d.getVar("GARAGE_SIGN_VERSION", True) or not d.getVar("SOTA_PACKED_CREDENTIALS", True):
+        return
+    import json
+    import urllib.request
+    import zipfile
+    with zipfile.ZipFile(d.getVar("SOTA_PACKED_CREDENTIALS", True), 'r') as zip_ref:
+        try:
+            with zip_ref.open('tufrepo.url', mode='r') as url_file:
+                url = url_file.read().decode() + '/health/version'
+        except (KeyError, ValueError, RuntimeError):
+            return
```

Comment:
Should we emit warnings in case of failures (both returns)?

Reply:
The various failure cases are:
* GARAGE_SIGN_VERSION is set. Not an error.
* SOTA_PACKED_CREDENTIALS is not set. Not an error.
* credentials.zip doesn't include tufrepo.url. Not an error.
* Something gets screwed up opening/reading the zip. This will presumably also break the other recipes that use the zip.
* The server returns something other than a 200. Presumably that means the server is broken and other things will break too.

Is it worth reporting a warning for those last two cases?